cli/command/service_test.go

```diff
@@ -0,0 +1,420 @@
+
+	base.Client.EXPECT().
+		ListServices().
+		Return([]*models.ServiceSummary{}, nil)
+
+	c := NewContext(t, nil, nil)
+
+	serviceCommand := NewServiceCommand(base.Command())
+	if err := serviceCommand.list(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestServiceLogs(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	query := url.Values{
+		"tail":  []string{"100"},
+		"start": []string{"start"},
+		"end":   []string{"end"},
+	}
+
+	base.Client.EXPECT().
+		ReadServiceLogs("svc_id", query).
+		Return([]*models.LogFile{}, nil)
+
+	args := Args{"svc_name"}
+	flags := Flags{"tail": 100, "start": "start", "end": "end"}
+	c := NewContext(t, args, flags)
+
+	serviceCommand := NewServiceCommand(base.Command())
+	if err := serviceCommand.logs(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestServiceLogs_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+	}
+
+	serviceCommand := NewServiceCommand(base.Command())
+	for name, c := range contexts {
+		t.Run(name, func(t *testing.T) {
+			if err := serviceCommand.logs(c); err == nil {
+				t.Fatal("%s: error was nil!", name)
+			}
+		})
+	}
+}
+
+func TestReadService(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Return(&models.Service{}, nil)
+
+	args := Args{"svc_name"}
+	c := NewContext(t, args, nil)
+
+	serviceCommand := NewServiceCommand(base.Command())
+	if err := serviceCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadService_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+	}
+
+	serviceCommand := NewServiceCommand(base.Command())
+	for name, c := range contexts {
+		t.Run(name, func(t *testing.T) {
+			if err := serviceCommand.read(c); err == nil {
+				t.Fatal("%s: error was nil!", name)
+			}
+		})
+	}
+}
+
+func TestScaleService(t *testing.T) {
+	testWaitHelper(t, func(t *testing.T, wait bool) {
+		defer client.SetTimeMultiplier(0)()
+
+		base, ctrl := newTestCommand(t)
+		defer ctrl.Finish()
+
+		base.Resolver.EXPECT().
+			Resolve("service", "svc_name").
+			Return([]string{"svc_id"}, nil)
+
+		scale := 2
+		req := models.UpdateServiceRequest{
+			ServiceID: "svc_id",
+			Scale:     &scale,
+		}
+
+		base.Client.EXPECT().
+			UpdateService(req).
+			Return("job_id", nil)
+
+		if wait {
+			job := &models.Job{
+				Status: "Completed",
+				Result: "svc_id",
+			}
+
+			base.Client.EXPECT().
+				ReadJob("job_id").
+				Return(job, nil)
+
+			deployments := []models.Deployment{
+				{
+					DesiredCount: 1,
```

Comment:
[nit] desired count above was 2 -- I know all that matters is that desired == running, but having 1 here instead of 2 may throw people off

Reply:
Nice catch ✅